test/e2e/certman_operator_tests.go

```diff
@@ -62,4 +80,186 @@ var _ = Describe("Certman Operator", Ordered, func() {
 			return apiserver.Spec.ServingCerts.NamedCertificates[0].ServingCertificate.Name == secretName
 		}, pollingDuration, 30*time.Second).Should(BeTrue(), "Certificate secret should be applied to apiserver object")
 	})
+
+	It("remove finalizer from the ClusterDeployment", func(ctx context.Context) {
+		clusterDeploymentGVR := schema.GroupVersionResource{
+			Group:    "hive.openshift.io",
+			Version:  "v1",
+			Resource: "clusterdeployments",
+		}
+
+		Eventually(func() bool {
+			cdList, err := dynamicClient.Resource(clusterDeploymentGVR).Namespace("certman-operator").List(ctx, metav1.ListOptions{})
+			if err != nil || len(cdList.Items) == 0 {
+				Fail("Error listing ClusterDeployments or no ClusterDeployment found")
+			}
+
+			clusterDeployment := cdList.Items[0]
+			cdName := clusterDeployment.GetName()
+			logger.Info("Processing ClusterDeployment", "name", cdName)
+
+			clusterDeployment.SetFinalizers([]string{})
+			_, err = dynamicClient.Resource(clusterDeploymentGVR).Namespace("certman-operator").Update(ctx, &clusterDeployment, metav1.UpdateOptions{})
+			if err != nil {
+				logger.Error(err, "Failed to remove finalizers from ClusterDeployment", "name", cdName)
+				return false
+			}
+
+			logger.Info("Finalizers removed from ClusterDeployment", "name", cdName)
+
+			updatedCD, err := dynamicClient.Resource(clusterDeploymentGVR).Namespace("certman-operator").Get(ctx, cdName, metav1.GetOptions{})
```

Comment:
As we discussed in one of the sync meet. Lets try to put the clusterdeployment into a separated namespace rather than `certman-operator`